### ISSUE TITLE
Create a narrow option for articles

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#2784]
# Story

As a website visitor I would like articles to have narrow columns so that it is easier to read and more in line with my expectations based on other sites.

This functionality was requested for the new Writing University website.

Example other sites: https://medium.com/mind-cafe/einsteins-formula-for-a-happy-life-b29aff61a9c7, https://www.nytimes.com/2021/01/04/technology/slack-down-outage.html

# Proposed solution
Create a new site-wide setting that applies to articles and periodicals to allow the site owner to display these in a narrow container.

**Comment by cory-skeers** [comment]
Interested in this. Also curious to look into some research and best practices surrounding this, might inform whether this should be a universal update or open to options. I've read some on it before, but don't know what kinds of margins there are around "optimum" widths for readability, if there's a clear best practice or if going off expectations is adequate.

**Comment by cory-skeers** [comment]
Sample resource from Smashing Magazine: https://www.smashingmagazine.com/2014/09/balancing-line-length-font-size-responsive-web-design/

**tl;dr version:** column width that accommodates around 45 to 85 characters (with the 65 characters as the optimum previously determined for print resources).

Grabbing random full-length lines from the examples:

https://medium.com/mind-cafe/einsteins-formula-for-a-happy-life-b29aff61a9c7: 77 characters,
https://www.nytimes.com/2021/01/04/technology/slack-down-outage.html: 65 characters,
https://sitenow.uiowa.edu/news/2020/12/changes-look-and-feel-person-page (unrestrained column size, so highly variable, but based on what my viewport happened to be upon loading): 140 characters

And the counterargument: http://web.archive.org/web/20170918212943/http://psychology.wichita.edu/surl/usabilitynews/72/LineLength.asp
Suggests that longer lines aren't as detrimental in digital media (they only tested up to 95 characters, though). Part of the reason the "optimum" above is bumped up to 85 rather than 75.

**Comment by joewhitsitt** [comment]
Related: https://github.com/uiowa/uiowa/issues/1609

**Comment by cory-skeers** [comment]
~Per backlog grooming discussion, looks like the plan is to go forward with adjusting the column width platform-wide for all articles using the current templates, prior to #1609 updates.~

**Comment by cory-skeers** [comment]
Scoped to Writing University only for now, with further discussion later.

**Comment by quamsta** [comment]
Might have made a duplicate issue here: https://github.com/uiowa/uiowa/issues/2924

But in general, this is something we'd like to see before migrating some Student Life sites with longform text/articles/posts. Ensuring readability by limiting the line length is something we think is really important across the board and all websites could benefit from it.

Providing an option to do this seems like a step in the right direction, but preventing extremely long lines of text automatically with no intervention from the user seems ideal to me.

**Comment by briand44** [comment]
The College of Pharmacy has interest in this.

**Comment by bspeare** [comment]
The featured image usages in the example narrow articles above (https://github.com/uiowa/uiowa/issues/2784#issue-778391839) seem like a good intersection with https://github.com/uiowa/uiowa/issues/2813.   

Maybe if the narrow option is turned on we alter the featured image to display the original image aspect-ratio within the featured image banner and render the banner as mobile stacked?


<img width="697" alt="Screen Shot 2022-11-22 at 1 50 12 PM" src="https://user-images.githubusercontent.com/1036433/203408023-b8497e6a-d0f9-4606-af84-d8701640e825.png">

**Comment by bspeare** [comment]
Here's another look at how changing the article body to narrow and the featured image treatment could work (my fonts aren't working in this screenshot):

<img width="1724" alt="Screen Shot 2022-11-22 at 2 10 14 PM" src="https://user-images.githubusercontent.com/1036433/203413100-cf6e2d3a-e8d0-4f7f-a33b-d505832435a0.png">

**Comment by cory-skeers** [comment]
Some more reading on this as a subject with links to various studies:
https://designregression.com/article/line-length-revisited-following-the-research

Some takeaways--the traditional wisdom of line lengths effecting reading speed have been challenged but not overturned. Newer research suggests it doesn't make as much (or maybe none) difference as a best practice, BUT that shorter line lengths may help with accessibility for some readers with particular reading difficulties, and that while longer lines may not technically make it more difficult to read, it often is judged as more difficult, longer time reading.

Fascinating stuff overall, but generally, best practice still seems to be narrow = better, just for different reasons than originally thought.

**Comment by pyrello** [comment]
Documenting here the result of #6042 and our subsequent discussion around it:
- @bspeare showed some examples of articles that would be affected. The primary concern is around floated images at certain sizes and how they would look if moved into a smaller container.
- @bspeare and @cory-skeers had previously discussed the possibility of using an update hook to switch image styles for large images or to remove the float on them. @cory-skeers did some probing in #6064 to see what this might look like. There were a lot of cases that were difficult to account for with either solution.
- We discussed creating a process to allow users to manually adjust their articles and switch over to the new format gradually versus "ripping off the bandage".
- It is our desire to eventually move to a situation where all articles are narrow, rather than an opt-in model.
- @joewhitsitt made the astute observation that people designing their articles for a desktop experience already are getting a different experience for mobile users than they might be expecting. 
- We discussed updating image size formats and/or implementing container query rules for image sizes. This seemed like a better investment than the work of a gradual rollout and would potentially mitigate some or most of the issues that might come up with switching the whole platform to narrow articles.
- Deadline for law journals is end of the month.

Based on the above, we came up with the following plan:
- Invest in container query work for image styles (issue to come) and possibly re-working image formats.
- Implement a narrow article feature and turn it on for the law journal sites (and possibly Writing University?) by the end of the month.
- After learning from the experience of the initial rollout, plan and communicate the rollout for the rest of the platform.
- Make any adjustments to narrow article styling, move it to the default platform, and remove the narrow article feature from sites it was set up on.

**Comment by bspeare** [comment]
Sondra and Jess provided some feedback on these narrow mockups and I have updated them to reflect their design suggestions:

- With featured image:  https://sandbox.prod.drupal.uiowa.edu/test-narrow-article-w-image
- Default article page design without featured image: https://sandbox.prod.drupal.uiowa.edu/test-narrow-article-w-gray-background

**Comment by quamsta** [comment]
@bspeare These look good. Would the featured image captions be placed similarly to how they are now?

**Comment by bspeare** [comment]
@quamsta I think they would remain the similar unless you see an opportunity to possibly adjust here?

**Comment by quamsta** [comment]
@bspeare Nothing too specific in mind, but just wanted to keep those shiny new captions in our thoughts as article layouts get adjusted. Something like this maybe?

<img width="1354" alt="Screenshot 2023-01-04 at 3 16 04 PM" src="https://user-images.githubusercontent.com/472923/210651543-e3978023-0d1d-4086-bc10-b73e985e4485.png">

**Comment by bspeare** [comment]
@quamsta That's a good suggestion.  I like the extra spacing there.

**Comment by briand44** [comment]
The slice of turning on the narrow container for Law Journals and possibly Writing University should implement a feature split since there are multiple sites. Include header area in scope of this based on mockups in https://github.com/uiowa/uiowa/issues/2784#issuecomment-1371369369. If we learn this is way too big once we get into it, lets discuss. Need to account for css differences in this feature, possibly preprocess similar to other examples. Large

**Comment by bspeare** [comment]
Since we have implemented a narrow article option, can we close this @briand44?

**Comment by briand44** [comment]
I didn't close this because we haven't rolled this out to the platform yet, which was what we had previously discussed at one point. Also, I see this issue is on the Grad College board for the Writing University so I wouldn't want to close it without feedback from @pyrello. I would be open to closing it and creating separate issues for Writing University and for the platform as a default/option if that makes more sense. I recently removed it from our project board assuming we weren't doing any work related to this in the short term.

**Comment by joewhitsitt** [comment]
Just read a long form article on a SiteNow site and as a user want to bump this issue up

**Comment by bspeare** [comment]
There is a related issue at https://github.com/uiowa/uiowa/issues/6342 where I propose turning on layout builder for articles and allowing only a few layouts and blocks.

**Comment by bspeare** [comment]
With https://github.com/uiowa/uiowa/issues/5426 coming up, I still like the original proposed solution here where you could adjust the articles to either be the default or narrow width.  Looking at the design pattern of the stories site, taking a story like https://stories.uiowa.edu/brandee-britt-social-media-strategy-hawkeye-athletics and putting it into a narrow format could look like https://sandbox.prod.drupal.uiowa.edu/news/2024/12/cubs-dreams-hawkeye-success.   In the sandbox example, we could set full images, blockquotes, and videos to bust out of the container to create a more dynamic article layout.
